src/vm/checker/typecheck/tests/mod.rs

```diff
@@ -286,6 +286,132 @@ fn test_factorial() {
     type_check(&":transient:", &mut contract, &mut analysis_db, false).unwrap();
 }
 
+#[test]
+fn test_set_variable() {
+    let contract_src = r#"
+        (define-data-var cursor int)
+        (define (get-cursor)
```

Comment:
Would it be more ergonomic to require an initial variable value to be set within the define statement? As in `(define-data-var cursor int 0)`.

Reply:
I like this idea. Maybe not as a requirement, but I'd like to be able to do that.
What's your take @kantai ?

Reply:
I’m not sure. It comes down to whether or not we expect there to be a common use case where someone would want to set a variable only after initialization from another contract (e.g., the rocket factory example.)

Though it may be nicer to be able to remove the optional return type in this case.

Reply:
If we added a “some” and “none” functions (and the requisite type specification parser), we’d get the best of both worlds.

So, one course of action is to require supplying a default in this PR (eliminating the need for the optional type in this version). Then, separately, once “some” and “none” are added, we’d be able to use them to define persisted variables as well.

Reply:
Addressed with Addressed with https://github.com/blockstack/blockstack-core/pull/1022/commits/9a5975e863eb7b086c5b5db8040bac9091b0cc24